### ISSUE TITLE
Cannot read property ‘registerBackButtonAction’ of undefined

### ISSUE THREAD

**Issue by chandanch** [driftyco/ionic#6935]
Hello, When I updated from ionic-angular2.0.0-beta.6 to 2.0.0-beta.7. I get this error : Cannot read property 'registerBackButtonAction' of undefined in app.js(line:31) on browser. i.e on node_modules/ionic-angular/components/app/app.js

I followed the steps given here: https://github.com/driftyco/ionic/blob/2.0/CHANGELOG.md#angular-update-to-200-rc1
Here is my package.json file:

```
{
  "dependencies": {
    "@angular/common": "^2.0.0-rc.2",
    "@angular/compiler": "^2.0.0-rc.2",
    "@angular/core": "^2.0.0-rc.2",
    "@angular/http": "^2.0.0-rc.2",
    "@angular/platform-browser": "^2.0.0-rc.2",
    "@angular/platform-browser-dynamic": "^2.0.0-rc.2",
    "@angular/router": "^3.0.0-alpha.6",
    "es6-shim": "^0.35.0",
    "ionic-angular": "^2.0.0-beta.7",
    "ionic-native": "^1.1.0",
    "ionicons": "3.0.0",
    "reflect-metadata": "^0.1.2",
    "rxjs": "^5.0.0-beta.6",
    "zone.js": "^0.6.12"
  },
  "devDependencies": {
    "del": "2.2.0",
    "gulp": "3.9.1",
    "gulp-watch": "4.3.5",
    "ionic-gulp-browserify-typescript": "^1.1.0",
    "ionic-gulp-fonts-copy": "^1.0.0",
    "ionic-gulp-html-copy": "^1.0.0",
    "ionic-gulp-sass-build": "^1.0.0",
    "ionic-gulp-scripts-copy": "^2.0.0",
    "run-sequence": "1.1.5"
  },
  "cordovaPlugins": [
    "cordova-plugin-device",
    "cordova-plugin-console",
    "cordova-plugin-whitelist",
    "cordova-plugin-splashscreen",
    "cordova-plugin-statusbar",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": [
    "ios",
    {
      "platform": "ios",
      "version": "",
      "locator": "ios"
    },
    "android"
  ],
  "name": "nanobi-analytics",
  "description": "Nanobi Analytics: An Ionic project"
}
```
My Current ionic version is: 2.0.0-beta.30

**Comment by manucorporat** [comment]
@chandanch please copy paste the output of `ionic info` and provide some source code to reproduce the problem! thanks

**Comment by chandanch** [comment]
Hello here is the ionic information:

Cordova CLI: Not installed
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.8
Ionic CLI Version: 2.0.0-beta.30
Ionic App Lib Version: 2.0.0-beta.16
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Mac OS X El Capitan
Node Version: v4.4.5
Xcode version: Xcode 7.3.1 Build version 7D1014

**Comment by chandanch** [comment]
This error is resolved now I updated to ionic-angular2.0.0.beta8 the main issue was I did not bootstrap the app component. I'm not getting this error now, But I'm getting a different error now:
**reflective_provider.js:163** 
Uncaught Invalid provider - only instances of Provider and Type are allowed, got: undefined

**Comment by jgw96** [comment]
Hello @chandanch great to hear that you fixed your first issue! Would you mind posting a plunker i can use to reproduce the second issue? Thanks for using Ionic!

**Comment by herbzi** [comment]
I also had this issue, I reverted to angular rc1 assuming that ionic doesn't support rc2 yet. I just changed "^2.0.0-rc.2" to "2.0.0-rc.1", deleted node_modules and reinstalled.

**Comment by herbzi** [comment]
I had the same reflective_provider error, I just assumed that ionic doesn't support angular rc2 yet, I changed "^2.0.0-rc.2" to "2.0.0-rc.1" deleted node_modules and reinstalled and it started working. I even have the same error on beta. 9 which I would have thought would support rc2.

**Comment by jgw96** [comment]
Hello all! This issue looks like it is because of a change in Angular 2 rc2. As rc2 was just released yesterday we are currently planning for the next release of ionic 2 to be on Angular 2 rc2. I will be keeping this issue open as a tracking issue. Thanks for using Ionic!

**Comment by chandanch** [comment]
Hello, Thanks @herbzi @jgw96 I was able to resolve the issue I changed it to **2.0.0-rc1** and it worked, Also change the **rxjs** version to "**5.0.0.beta-6**".

Here is my updated **package.json** file for reference:

`{
  "dependencies": {
    "@angular/common": "^2.0.0-rc.1",
    "@angular/compiler": "^2.0.0-rc.1",
    "@angular/core": "^2.0.0-rc.1",
    "@angular/http": "^2.0.0-rc.1",
    "@angular/platform-browser": "^2.0.0-rc.1",
    "@angular/platform-browser-dynamic": "^2.0.0-rc.1",
    "@angular/router": "^2.0.0-rc.1",
    "es6-shim": "^0.35.0",
    "ionic-angular": "^2.0.0-beta.8",
    "ionic-native": "^1.1.0",
    "ionicons": "3.0.0",
    "reflect-metadata": "^0.1.2",
    "rxjs": "^5.0.0-beta.6",
    "zone.js": "^0.6.12"
  },
  "devDependencies": {
    "del": "2.2.0",
    "gulp": "3.9.1",
    "gulp-watch": "4.3.5",
    "ionic-gulp-browserify-typescript": "^1.1.0",
    "ionic-gulp-fonts-copy": "^1.0.0",
    "ionic-gulp-html-copy": "^1.0.0",
    "ionic-gulp-sass-build": "^1.0.0",
    "ionic-gulp-scripts-copy": "^2.0.0",
    "run-sequence": "1.1.5"
  },
  "cordovaPlugins": [
    "cordova-plugin-device",
    "cordova-plugin-console",
    "cordova-plugin-whitelist",
    "cordova-plugin-splashscreen",
    "cordova-plugin-statusbar",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": [
    "ios",
    {
      "platform": "ios",
      "version": "",
      "locator": "ios"
    },
    "android"
  ],
  "name": "nanobi-analytics",
  "description": "Nanobi Analytics: An Ionic project"
}
`

**Comment by jgw96** [comment]
Awesome! Glad to hear that fixed it for you. Thanks for using Ionic!

**Comment by russcarver** [comment]
This did not work for me in upgrading from Ionic2 beta.6 to Ionic2 beta.9.  I tried Angular2 rc.2 and rc.1 both to no avail.  I tried removing the browser platform, re-adding and building it as well as deleting node modules each time.  I even tried beta.8.

I had to start a new Ionic2 project and copy over the changed parts of package.json, config.xml, index.html and app.ts.  Notable, in particular, is the removal of @App and @Page - replaced with @Component.  This is not part of the linked instructions above.
